src/ui/public/validate_query/validate_query.js

```diff
@@ -32,40 +32,37 @@ define(function (require) {
           };
 
           function validator(query) {
-            var index;
-            var type;
             if (request.abort) request.abort();
```

Comment:
In order for request.abort to exist it needs to be the exact promise that was returned by the es client. These changes set request on line 39 instead, so `request.abort()` will never be defined.